### DISCUSSION TITLE
How to access the current record in table actions

### DISCUSSION THREAD

**Discussion by Xoshbin** [filamentphp/filament#3961] [Help]
How can I access the current record in actions please?
```
protected function getTableQuery(): Builder
    {
        return Auth::user()->followers()->getQuery();
    }
protected function getTableActions(): array
    {
        return [
            Action::make('unfollow')
                ->action(fn(Followable $record) => $this->unfollow($record)),
        ];
    }
public function unfollow($record)
    {
        Auth::user()->unfollow($record);
    }
```
it gives this error:

App\Http\Livewire\Dashboard\Followers::App\Http\Livewire\Dashboard{closure}(): Argument #1 ($record) must be of type Overtrue\LaravelFollow\Followable, null given, called in /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php on line 36

I made a repo to reproduce the issue.

https://github.com/Xoshbin/filament-action-record

**Comment by RX-77IT** [comment]
I also had this problem today because I still get confused when I'm on a Page or on a $table. 

I then found [this thread](https://github.com/filamentphp/filament/discussions/4648) and understood this:

- when on a Page (like EditModel.php) $record is null, so to get the current record one has to use `$this->getRecord()`. In your case I think this should work
```php
protected function getTableActions(): array
{
    return [
        Action::make('unfollow')
            ->action(fn() => $this->unfollow($this->getRecord())),
    ];
}
```
- when in a $table one can pass the $record as a parameter of the Closure, like what you and I were trying to do.


EDIT: or maybe I completely misinterpret the problem because i saw now that you are using `getTableActions()` and that's a table method.

**Reply by danharrin** [reply]
The current table row record is `$record`, on the edit page the record you're editing is `$this->getRecord()`

**Reply by lourbert** [reply]
can I use the getRecord() to custom page of the same resource to get the record id

**Reply by leandrocfe** [reply]
Use [InteractsWithRecord](https://filamentphp.com/docs/3.x/panels/resources/custom-pages#using-a-resource-record)
